server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -580,7 +580,8 @@ static long fallbackRegionSize(JvmInfo jvmInfo) {
             } else if (regionSize > ByteSizeUnit.MB.toBytes(32)) {
                 regionSize = ByteSizeUnit.MB.toBytes(32);
             }
-            return regionSize;
+            // elasticsearch overrides the region size for small heaps so the smallest region size possible is 4MiB
+            return Math.max(ByteSizeUnit.MB.toBytes(4), regionSize);
```

Comment:
If we had overridden the region size, we should see it in the jvm options jvmInfo has (it just looks them up essentially from the command line through the diagnostic mx bean). Rather than hardcoding this (which could get out of sync), I wonder why we are having to fallback at all. In what circumstances is the region size not present?

Reply:
I think the defensiveness here:

https://github.com/elastic/elasticsearch/blob/19257125b19ff05a6b014f37693f9417ca579626/server/src/main/java/org/elasticsearch/monitor/jvm/JvmInfo.java#L123

caused us to consider this necessary. I suppose `HotSpotDiagnosticMXBean` is also not necessarily supported everywhere in theory. I also wonder if we can strengthen this nowadays?